pyproject.toml

```diff
@@ -201,35 +201,35 @@ max-line-length = 105  # default 100
 [tool.pylint."messages control"]
 disable = [
 # intentionally disabled:
-    "spelling",  # too noisy
+    "arguments-renamed", # mostly used in the transpiler, renaming seems natural
```

Comment:
Similar to the `Options` one - if this is just in one place (and it's happening there just for backwards compatibility, iirc), could we enable the lint and put a local skip on it?

In general this is a good lint, I think - for better or worse, Python allows arguments to be passed as keywords by default, so the subclass relationship largely should respect argument names.

Reply:
ready as well